adala/agents/base.py

```diff
@@ -216,6 +216,16 @@ def get_teacher_runtime(self, runtime: Optional[str] = None) -> Runtime:
             )
         return runtime
 
+    def get_skills(self) -> List[Skill]:
```

Comment:
why do we need this function? Skills must be always Dict mapping from skill name to Skill (there is a `skills_validator` function to prevalidate that), so we can always iterate like
```
for skill_name in agent.skills.get_skill_names():
     skill = agent.skills[skill_name]
```

Reply:
> so we can't count on this always being a SkillSet 
we can because of https://github.com/HumanSignal/Adala/blob/master/adala/agents/base.py#L109

Reply:
Can we change the type sig of `skills` in `Agent` then?
And still, like I said in my first comment, it seems simpler to have a `SkillSet.get_skills` than `SkillSet.get_skill_names`, esp since `SkillSet.get_skill_names` doesn't seem to be used anywhere. I'll go ahead and change that and the type sig of `Agent.skills`